packages/deployments/contracts/contracts/messaging/connectors/SpokeConnector.sol

```diff
@@ -591,4 +624,14 @@ abstract contract SpokeConnector is Connector, ConnectorManager, WatcherClient,
     // emit process results
     emit Process(_messageHash, _success, _returnData);
   }
+
+  /**
+   * @notice This function calculates the last completed snapshot id
+   * @return _lastCompletedSnapshotId The last completed snapshot id
+   */
+  function _getLastCompletedSnapshotId() internal view returns (uint256 _lastCompletedSnapshotId) {
```

Comment:
should we put this function in a library or something that both the root manager and spoke connector can use?

Reply:
Will do